### ISSUE TITLE
Test uses obsolete syntax

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/sdk#28362]
The test language/mixin_illegal_syntax_test.dart uses an old style of syntax in which a `typedef` is used to create a mixin application. I think we should remove tests that test obsolete features.

**Comment by eernstg** [comment]
I guess the **point** is that all the tested tools are forced to enforce that the given obsolete feature is rejected. Otherwise some tools could allow it, and developers "out there" could keep using them.

After a while, some new syntax could be introduced which happens to give some usages of the old feature a new meaning (maybe the syntactic structure is different, but the point is that it is now intended to be OK to write something that looks like a usage of an obsolete feature).

At that point, obviously, we can't keep a test which will require some (maybe even all) legitimate usages of that new feature to be rejected. So we have to **remove** tests enforcing that an obsolete feature is rejected "if/**when** that **syntax is reused**".

Until such a thing happens, however, I do not think it is important (maybe not even a good idea) to get rid of this type of test. Just like a bunch of 'regress_*_test.dart', it may be very unlikely that the same bug comes back, but we still keep these tests indefinitely, just in case. I don't see a compelling reason why we shouldn't do the same thing for these reject-obsolete-feature tests.

In the concrete example, `tests/language/mixin_illegal_syntax_test.dart`, I believe that we have no plans to introduce syntax that actually overlaps with the old mixin syntax (we are discussing `typedef T = List<int>;` and stuff like that, but I believe that nobody has argued for reintroducing `typedef T = C with M;`).

In summary, I wouldn't have a problem with a general rule that says "after a while, remove tests that enforce rejection of obsolete features", but I don't actually see a strong motivation for doing it.
